docs/4.x/archiving-behavior-specification.md

```diff
@@ -0,0 +1,508 @@
+  to initiate archiving for all plugins.
+  
+  Note: for some archives (where period=range or if using a custom segment), we will only initiate archiving for the requested plugin(s).
+- **Piwik\PluginsArchiver**: This class will loop through every plugin (or requested plugin), create the plugin's `Archiver` instance
+  and invoke it. Based on whether a day or non-day period is being archived, either the `aggregateDayReports()` or `aggregateMultipleReports()`
+  method will be invoked.
+- **Piwik\Plugin\Archiver**: This is the base class for all plugin archivers. Plugins define their archiving logic within one of these
+  classes, which will then be invoked by PluginsArchiver.
+- **Piwik\ArchiveProcessor**: This is a utility class that contains methods to easily query and aggregate metrics and reports for multiple periods.
+  It is mostly used by `aggregateMultipleReports()` implementations to aggregate and insert archive data with a single method call, and by
+  both `aggregateMultipleReports()` and `aggregateDayReports()` to insert archive data.
+- **Piwik\DataAccess\LogAggregator**: This is a utility class that contains some methods to aggregate log data. It is mostly used by
+  `aggregateDayReports()` implementations to aggregate log data and turn the result into `DataTable`s to insert. Many plugin may find
+  that those methods are not enough, however, and will query log tables directly.
+- **Piwik\DataAccess\ArchiveWriter**: This class contains low level methods to create, write to and delete archives. It is used by
+  `ArchiveProcessor` to initialize new archives, insert archive data into them, and to finalize them when finished.
+- **Piwik\Archive\Parameters**: This is a utility class that contains all the parameters of an archive data queries, such as the
+  site IDs to get data for, the periods to get data for and the segment to get data for.
+- **Piwik\ArchiveProcessor\Parameters**: This is a utility class that, like `Piwik\ArchiveParameters`, contains the parameters for
+  a single archiving run. It only stores one site ID and period, however, since we can only launch the archiving process for a single site/period
+  at a time. It also contains some other information like the request plugin/report if any or whether we want to only archive data for
+  a single report.
+
+The entire code path for Core Archiving is as follows:
+
+- An API class constructs an `Archive` instance and queries for archive data.
+- If archiving is not allowed to launch, `Archive` will look through the archive tables directly for data and return whatever it finds.
+- If archiving is allowed, `Archive` will initiate archiving for every site/period combination in the query. It does this by calling the
+  `CoreAdminHome.archiveReports` API method.
+- `CoreAdminHome.archiveReports` creates and invokes the `Loader` class.
+- `Loader` looks for a usable archive. If found, it returns that archive ID. Otherwise it launches the archiving process, first aggregating
```

Comment:
```suggestion
- `Loader` looks for a usable archive. If found, it returns that archive ID. Otherwise, it launches the archiving process, first aggregating
```